### ISSUE TITLE
Database.DropAsync() Does Not Work!!!

### ISSUE THREAD

**Issue by digitalkingsley** [coronabytes/dotnet-arangodb#28]
In my asp.net core project, after creating a context with the **IArangoContext** interface, I'm able to successfully create a database with the following code:

`_arango.Database.CreateAsync("dbname")` - **_arango** is the **IArangoContext** context used to create the database.

However, when I attempt to drop the created database (with DropAsync), nothing happens. Here's a sample code I wrote to attempt a drop:

`await _arango.Database.DropAsync("dbname");`  

In fact, when I use the following code:

`bool dbExists = _arango.Database.ExistAsync()` to check if the database I created exists before attempting to drop, it returns a "false"; meaning that the database doesn't exist. However, when I fire-up **arangosh** and issue the command:

`_db.databases()`, it fetches all the databases, including the one the above C# code claimed is non-existent.

Finally, I'm able to drop the database via **arangosh** using the command `_db.dropDatabase("dbname") `and it successfully drops the database. But the `_arango.Database.DropAsync("dbname");` code does nothing to the database; neither does the bool `dbExists = _arango.Database.ExistAsync() `code work as expected.

Is this a known-issue? Or am I missing something?

**Comment by digitalkingsley** [comment]
Hi,

Yes, I'm using a realm. Here's my connection string (I've replaced the User and Password):

`"Fake": "Server=http://localhost:8529; Realm=Mns; User=foo; Password=fie;"`

The Database name is: **testdb**

But of course, once the database is created, I noticed it prepends the the database-name with the realm, i.e. **Mns-testdb**, in my case. So, when I attempt the drop from my C# code, I use the new name **Mns-testdb**, but it says the database doesn't exist. Also, when I try to use the original database-name (**testdb**), it doesn't work.

Attempting the drop using the **testdb** name in **arangosh** doesn't also work (which should be expected, because the database in the server is now prepended with the realm), but using the **Mns-testdb** name works.

Kindly note that I'm running Arangodb in a docker container.

**Comment by coronabytes** [comment]
works for me,
I have a 3 db-node cluster in docker under wsl2

```csharp
[Fact]
public async Task Create2()
{
    var arango = new ArangoContext("Server=http://localhost:8529; Realm=Mns; User=root; Password=;");
    await arango.Database.CreateAsync("testdb");
    Assert.True(await arango.Database.ExistAsync("testdb"));
    await arango.Database.DropAsync("testdb");
    Assert.False(await arango.Database.ExistAsync("testdb"));
}
```

**Comment by digitalkingsley** [comment]
I'm using the latest driver. I'm also using the latest arangodb image, pulled from docker hub.

You may not be able to reproduce my problem with what you have done. You have just shown me a test-case which doesn't necessarily mimic the environment under which I'm running and encountering the problem.

Can you create a simple asp.net core Web API project, register arangodb in Startup.cs (according to the **readme**), inject the IArangoContext in a controller using constructor-injection (again, according to the **readme**), then use the context to attempt to create and drop a database. The creation works, but the drop does not. The ExistAsync() also lies, by returning a "false" when it should be returning "true".

**Comment by digitalkingsley** [comment]
Hi,

Thanks for your help. I've gotten it to work. I took a close-look at your test and noticed that you're not prepending the database-name with the realm, when passing it as an argument to the CreateAsync() and ExistAsync() methods. I had tried that initially, but it didn't work. Now, after restarting my container and machine and trying again, everything works as expected. 

Thanks a lot for your help. I really appreciate.
